sample_size/metrics.py

```diff
@@ -60,6 +67,15 @@ def _check_probability(probability: float) -> float:
         else:
             raise ValueError("Error: Please provide a float between 0 and 1 for probability.")
 
+    def generate_p_value(self, true_alt: bool, size: int, variants: int) -> Any:
```

Comment:
can we avoid Any as the return type? same for other ones

Reply:
Originally I used `Union[float, List[float]]` as all `generate_p_value` return type, but I always got the following type check error:
`Returning Any from function declared to return "Union[float, List[float]]"  [no-any-return]`

the actual return type of this function is either one p-value(when variants=2) or a list of p-values

Reply:
the problem is probably because np function didn't indicate a type so mypy doesn't know its type. you can explicitly provide it e.g.

```python
    def generate_p_value(self, true_alt: bool, size: int, variants: int) -> Union[np.typing.NDArray[np.float64], float]:
        if not true_alt:
            return np.array(stats.uniform.rvs(0, 1, size=variants - 1))

        else:
            nc = np.sqrt(size / 2) * self.mde / self.variance
            z_alt = stats.nct.rvs(nc=nc, df=2 * (size - 1), size=variants - 1)
            return float(2 * stats.norm.sf(np.abs(z_alt)))
```

and update your test to be 
```python
from numpy.testing import assert_array_equal
assert_array_equal(p, pval)
```